sql-common/my_time.c

```diff
@@ -336,12 +339,25 @@ str_to_datetime(const char *str, uint length, MYSQL_TIME *l_time,
     pos++;
     skip_digits(&pos, end); // ignore the return value
   }
+  if (pos < end && *pos == '-' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu-HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
+  }
+  else if (pos < end && *pos == '+' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu+HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
```

Comment:
There is already some behaviour related to existing timezones with time parsing that need to preserved in 10.1 and 10.2.

For times like rfc3339/iso8601 with a timezone suggest using the specified timezone as it is a new behaviour.

The decision on 10.1 or 10.2 isn't mine to make however I think 10.2 might be a better target.